docs/content/advanced/backend/payment/how-to-create-payment-provider.md

```diff
@@ -53,51 +53,44 @@ For example, create the file `src/services/my-payment.ts` with the following con
 <!-- eslint-disable max-len -->
 
 ```ts title=src/services/my-payment.ts
-import { 
-  AbstractPaymentService, 
-  Cart, Data, Payment, PaymentSession, 
-  PaymentSessionStatus, TransactionBaseService, 
-} from "@medusajs/medusa"
+import { AbstractPaymentService, Context, Data, Payment, PaymentSession, PaymentSessionStatus, TransactionBaseService } from "@medusajs/medusa"
 import { EntityManager } from "typeorm"
 
 class MyPaymentService extends AbstractPaymentService<TransactionBaseService> {
   protected manager_: EntityManager
   protected transactionManager_: EntityManager
 
-  getPaymentData(paymentSession: PaymentSession): Promise<Data> {
+  async getPaymentData(paymentSession: PaymentSession): Promise<Data> {
     throw new Error("Method not implemented.")
   }
-  updatePaymentData(paymentSessionData: Data, data: Data): Promise<Data> {
+  async updatePaymentData(paymentSessionData: Data, data: Data): Promise<Data> {
     throw new Error("Method not implemented.")
   }
-  createPayment(cart: Cart): Promise<Data> {
+  async createPayment(context: Context): Promise<PaymentSessionResponse> {
     throw new Error("Method not implemented.")
   }
-  retrievePayment(paymentData: Data): Promise<Data> {
+  async retrievePayment(paymentData: Data): Promise<Data> {
     throw new Error("Method not implemented.")
   }
-  updatePayment(paymentSessionData: Data, cart: Cart): Promise<Data> {
+  async updatePayment(paymentSessionData: Data, context: Context): Promise<PaymentSessionResponse> {
     throw new Error("Method not implemented.")
   }
-  authorizePayment(
-    paymentSession: PaymentSession,
-    context: Data
-  ): Promise<{ data: Data; status: PaymentSessionStatus; }> {
+  async authorizePayment(paymentSession: PaymentSession, context: Data): Promise<{ data: Data; status: PaymentSessionStatus; }> {
     throw new Error("Method not implemented.")
   }
-  capturePayment(payment: Payment): Promise<Data> {
+  async capturePayment(payment: Payment): Promise<Data> {
     throw new Error("Method not implemented.")
   }
-  refundPayment(payment: Payment, refundAmount: number): Promise<Data> {
+  async refundPayment(payment: Payment, refundAmount: number): Promise<Data> {
```

Comment:
todo: `: Promise<PaymentData>`

Reply:
From looking at the payment provider interface, I can see that Data = PaymentSessionData = PaymentData. I think, unless there's a reason I'm not aware of, we should default to one of them in the documentation. Which one do you think we should choose?

Reply:
@adrien2p just in case you missed this comment 👆🏻

Reply:
Oups sorry, indeed they are the same. We did split them in order to make sense of what type of object it correspond to. I am fine with choosing one and in that case we can stick to data as you did. I am just afraid that the user don't understand why we say data but the interface says something else

Reply:
Hmm, that makes sense too. I'll look into it and see the best approach here.